src/libexpr/flake/call-flake.nix

```diff
@@ -75,11 +79,27 @@ let
 
         in
           if node.flake or true then
-            assert builtins.isFunction flake.outputs;
-            result
+            { 
+              inherit flake result;
+              cacheKey = "${builtins.toJSON [sourceInfo.narHash inputCacheKey]}";
```

Comment:
Probably wanna use something better in the final version, but this works for the PoC.